contrib/storage-hive/core/src/main/java/org/apache/drill/exec/store/hive/HiveUtilities.java

```diff
@@ -507,5 +510,52 @@ public static boolean hasHeaderOrFooter(HiveTableWithColumnCache table) {
     int skipFooter = retrieveIntProperty(tableProperties, serdeConstants.FOOTER_COUNT, -1);
     return skipHeader > 0 || skipFooter > 0;
   }
+
+  /**
+   * This method checks whether the table is transactional and set necessary properties in {@link JobConf}.
+   * If schema evolution properties aren't set in job conf for the input format, method sets the column names
+   * and types from table/partition properties or storage descriptor.
+   *
+   * @param job the job to update
+   * @param properties table or partition properties
+   * @param sd storage descriptor
+   */
+  public static void verifyAndAddTransactionalProperties(JobConf job, Properties properties, StorageDescriptor sd) {
```

Comment:
Is it necessary to pass both `JobConf` and `Properties`? As far as I can see `job` is always populated using passed `properties`.